rockcraft/oci.py

```diff
@@ -0,0 +1,205 @@
+        """
+        src_path = self.path / self.image_name
+        dest_path = image_dir / image_name
+        _copy_image(f"oci:{str(src_path)}", f"oci:{str(dest_path)}")
+
+        return Image(image_name=image_name, path=image_dir)
+
+    def extract_to(self, bundle_dir: Path) -> Path:
+        """Unpack the image to an OCI runtime bundle.
+
+        :param bundle_dir: The directory to store runtime bundles.
+        """
+        ui.emit.progress(f"Extracting {self.image_name}")
+
+        bundle_dir.mkdir(parents=True, exist_ok=True)
+        bundle_path = bundle_dir / self.image_name.replace(":", "-")
+        image_path = self.path / self.image_name
+        shutil.rmtree(bundle_path, ignore_errors=True)
+        _process_run(["umoci", "unpack", "--image", str(image_path), str(bundle_path)])
+
+        ui.emit.message(f"Extracted {self.image_name}")
+
+        return bundle_path / "rootfs"
+
+    def add_layer(self, tag: str, layer_path: Path) -> None:
+        """Add a layer to the image.
+
+        :param tag: The tag of the image containing the new layer.
+        :param layer_path: The path to the new layer root filesystem.
+        """
+        image_path = self.path / self.image_name
+
+        ui.emit.progress("Creating new layer")
+
+        temp_file = Path(self.path, f".temp_layer.{os.getpid()}.tar")
+        temp_file.unlink(missing_ok=True)
+
+        try:
+            old_dir = os.getcwd()
+            try:
+                os.chdir(layer_path)
+                with tarfile.open(temp_file, mode="w") as tar_file:
+                    for root, _, files in os.walk("."):
+                        for name in files:
+                            path = os.path.join(root, name)
+                            ui.emit.trace(f"Adding to layer: {path}")
+                            tar_file.add(path)
+            finally:
+                os.chdir(old_dir)
+
+            _process_run(
+                [
+                    "umoci",
+                    "raw",
+                    "add-layer",
+                    "--tag",
+                    tag,
+                    "--image",
+                    str(image_path),
+                    str(temp_file),
+                ]
+            )
+        finally:
+            temp_file.unlink(missing_ok=True)
+
+        ui.emit.message("Created new layer")
+
+    def to_docker_daemon(self, tag: str) -> None:
+        """Export the current image to the local docker daemon.
+
+        :param tag: The tag to export.
+        """
+        parts = self.image_name.split(":", 1)
+        name = parts[0]
+
+        ui.emit.progress(f"Exporting {name}:{tag} to local docker daemon")
+
+        src_path = self.path / f"{name}:{tag}"
+        _copy_image(f"oci:{str(src_path)}", f"docker-daemon:{name}:{tag}")
+
+        ui.emit.message(f"Exported {name}:{tag} to local docker daemon")
+
+    def to_oci_archive(self, tag: str) -> None:
+        """Export the current image to the local docker daemon.
```

Comment:
this docstring is the same as the one above, what is the difference (can you add that to the doc?)

Reply:
A result of the magic of copy and paste. Fixing.